lib/Runtime/Library/JsBuiltInEngineInterfaceExtensionObject.cpp

```diff
@@ -453,5 +460,34 @@ namespace Js
         return obj;
     }
 
+    Var JsBuiltInEngineInterfaceExtensionObject::EntryJsBuiltIn_Internal_ArraySpeciesCreate(RecyclableObject *function, CallInfo callInfo, ...)
+    {
+        EngineInterfaceObject_CommonFunctionProlog(function, callInfo);
+        Assert(args.Info.Count == 3);
+
+        bool isBuiltinArrayCtor = true;
+        RecyclableObject * newObj = JavascriptArray::ArraySpeciesCreate(args.Values[1], 0, scriptContext, nullptr, nullptr, &isBuiltinArrayCtor);
+        JavascriptArray* newArr = nullptr;
+
+        if (newObj == nullptr)
+        {
+            newArr = scriptContext->GetLibrary()->CreateArray(0);
+            newObj = newArr;
+        }
+        else
+        {
+            // If the new object we created is an array, remember that as it will save us time setting properties in the object below
+            newArr = JavascriptOperators::TryFromVar<JavascriptArray>(newObj);
+            if (newArr)
+            {
+#if ENABLE_COPYONACCESS_ARRAY
+                JavascriptLibrary::CheckAndConvertCopyOnAccessNativeIntArray<Var>(newArr);
+#endif
+            }
+        }
+
+        return newObj;
```

Comment:
Should this be returning `newArr`? if not, whats the point of `newArr`?

Reply:
This code is essentially copied from JavascriptArray::FilterHelper. I don't have a good understanding of under what circumstances ArraySpeciesCreate would return a JavascriptCopyOnAccessNativeIntArray that needs conversion so I left that part of the logic the same. On closer inspection though, yeah, newArr is redundant and unnecessary.